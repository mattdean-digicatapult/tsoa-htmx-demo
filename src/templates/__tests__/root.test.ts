import { describe, expect, it } from 'bun:test'

import RootTemplates from '../root'

describe('RootTemplates', () => {
  describe('Root', () => {
    it('should render root page', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Root('title')
      expect(rendered).toMatchSnapshot()
    })

    it('should escape html in title', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Root('<div>Malicious Content</div>')
      expect(rendered).toMatchSnapshot()
    })

    it('should render counter 1 on second load', async () => {
      const templates = new RootTemplates()
      await templates.Root('title')
      const rendered = await templates.Root('title')
      expect(rendered).toMatchSnapshot()
    })
  })

  describe('Counter', () => {
    it('should render the counter with value 0', async () => {
      const templates = new RootTemplates()
      const rendered = await templates.Counter()
      expect(rendered).toMatchSnapshot()
    })

    it('should render the counter on second call with value 1', async () => {
      const templates = new RootTemplates()
      await templates.Counter()
      const rendered = await templates.Counter()
      expect(rendered).toMatchSnapshot()
    })
  })
})
