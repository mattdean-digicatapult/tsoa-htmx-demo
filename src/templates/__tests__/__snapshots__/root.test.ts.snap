// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`RootTemplates Root should render root page 1`] = `"<!DOCTYPE html><html><head><script src="lib/htmx.org/htmx.min.js"></script><link rel="icon" type="image/ico" sizes="48x48" href="/public/images/favicon.ico"/><link rel="stylesheet" type="text/css" href="/public/styles/main.css"/><title>title</title></head><body><div id="counter" hx-get="/counter" hx-trigger="every 1s" hx-swap="outerHTML">0</div></body></html>"`;

exports[`RootTemplates Root should render counter 1 on second load 1`] = `"<!DOCTYPE html><html><head><script src="lib/htmx.org/htmx.min.js"></script><link rel="icon" type="image/ico" sizes="48x48" href="/public/images/favicon.ico"/><link rel="stylesheet" type="text/css" href="/public/styles/main.css"/><title>title</title></head><body><div id="counter" hx-get="/counter" hx-trigger="every 1s" hx-swap="outerHTML">1</div></body></html>"`;

exports[`RootTemplates Counter should render the counter with value 0 1`] = `"<div id="counter" hx-get="/counter" hx-trigger="every 1s" hx-swap="outerHTML">0</div>"`;

exports[`RootTemplates Counter should render the counter on second call with value 1 1`] = `"<div id="counter" hx-get="/counter" hx-trigger="every 1s" hx-swap="outerHTML">1</div>"`;

exports[`RootTemplates Root should escape html in title 1`] = `"<!DOCTYPE html><html><head><script src="lib/htmx.org/htmx.min.js"></script><link rel="icon" type="image/ico" sizes="48x48" href="/public/images/favicon.ico"/><link rel="stylesheet" type="text/css" href="/public/styles/main.css"/><title>&lt;div&gt;Malicious Content&lt;/div&gt;</title></head><body><div id="counter" hx-get="/counter" hx-trigger="every 1s" hx-swap="outerHTML">0</div></body></html>"`;
